app/services/WhatYouOweService.scala

```diff
@@ -76,7 +77,7 @@ class WhatYouOweService @Inject()(val financialDetailsService: FinancialDetailsS
 
         val whatYouOweChargesList = WhatYouOweChargesList(
           balanceDetails = balanceDetails,
-          chargesList = getFilteredChargesList(financialDetailsModelList),
+          chargesList = getFilteredChargesList(financialDetailsModelList)(dateService),
```

Comment:
no need to pass implicit dateService on this occasion, as dateService in a global value within this class as you can specify this it directly  on line 106 when you calling getAllDocumentDetailsWithDueDates